diff -Naur R-2.8.1/src/unix/system.c R-2.8.1.tpg/src/unix/system.c
--- R-2.8.1/src/unix/system.c	2008-09-21 22:05:06.000000000 +0000
+++ R-2.8.1.tpg/src/unix/system.c	2008-12-22 20:31:37.000000000 +0000
@@ -294,6 +294,9 @@
 #ifdef HAVE_TCLTK
     if(useTk) {
 	R_GUIType = "Tk";
+	/* If we are started by a menu, we don't get a tty, 
+	* but _are_ interactive */
+	R_Interactive = TRUE;
     }
 #endif
     R_common_command_line(&ac, av, Rp);
