--- R-3.0.0/src/library/tcltk/R/tkGUI.R.orig	2013-04-07 22:26:29.842692112 +0100
+++ R-3.0.0/src/library/tcltk/R/tkGUI.R	2013-04-07 22:27:11.536689676 +0100
@@ -60,6 +60,8 @@
           command = quote(demo(tkcanvas)))
     tkadd(demoMenu, "command", label = gettext("R FAQ"),
           command = quote(demo(tkfaq)))
+    tkadd(demoMenu,"command",label = gettext("Graphics"),
+          command = quote(demo(graphics)))
 
     loadpackageWidget <- function()
     {
